lib/WorseReflection/Bridge/TolerantParser/Diagnostics/WrongVarOrderProvider.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Phpactor\WorseReflection\Bridge\TolerantParser\Diagnostics;
+
+use Generator;
+use Microsoft\PhpParser\Node;
+use Phpactor\DocblockParser\Ast\Tag\InvertedVarTag;
+use Phpactor\DocblockParser\DocblockParser;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\WorseReflection\Core\DiagnosticProvider;
+use Phpactor\WorseReflection\Core\Inference\Frame;
+use Phpactor\WorseReflection\Core\Inference\NodeContextResolver;
+
+class WrongVarOrderProvider implements DiagnosticProvider
+{
+    private DocblockParser $docblockParser;
+
+    public function __construct(DocblockParser $docblockParser = null)
```

Comment:
this should be fine, we can refactor to inject the [CachedDocblockParser](https://github.com/phpactor/phpactor/blob/581bd4dc40250fcb2f95d1428ff1790e1ee74a21/lib/WorseReflection/Core/ServiceLocator.php#L98-L101)

caching is pretty crucial here